composer/models/huggingface.py

```diff
@@ -302,6 +300,14 @@ def eval_forward(self, batch, outputs: Optional[Any] = None):
             # if we are in the single class case, then remove the classes dimension
             if output.shape[1] == 1:
                 output = output.squeeze(dim=1)
+        else:
+            output = outputs if outputs else self.forward(batch)
+            self.labels = batch.pop('labels')
```

Comment:
I think we don't want to pop the labels in the else case here? to match the previous behavior

Reply:
I think you added this back in somehow?